controller/src/main/java/cern/jarrace/controller/rest/controller/AgentContainerController.java

```diff
@@ -0,0 +1,104 @@
+package cern.jarrace.controller.rest.controller;
+
+import cern.jarrace.controller.domain.Entrypoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * @author timartin
+ * Controller that exposes services to manage {@link Entrypoint}s
+ */
+@RestController
+@RequestMapping("/jarrace")
+public class AgentContainerController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AgentContainerController.class);
+    private static final File DEPLOYMENT_DIR = new File(System.getProperty("java.io.tmpdir"));
+
+    final Map<String, List<Entrypoint>> entrypoints = new HashMap<>();
+    final Map<String, String> paths = new HashMap<>();
+
+    @RequestMapping(value = "/container/deploy/{name}", method = RequestMethod.POST)
+    public void deploy(@PathVariable("name") String name, @RequestBody byte[] jar) throws IOException {
+        System.out.println("Deployed " + name);
+        String path = writeFile(name, jar);
+        paths.put(name, path);
+        entrypoints.put(name, new ArrayList<>());
+        List<String> args = new ArrayList<>();
+        args.add("cern.jarrace.agent.AgentContainer");
+        args.add(name);
+        args.add("localhost:8080");
+        startContainer(path, args);
+    }
+
+    @RequestMapping(value = "/service/register/{name}", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void registerService(@PathVariable(value = "name") String name, @RequestBody Entrypoint entrypoint){
+
+        if(entrypoint.getAgentName() == null || entrypoint.getAgentName().isEmpty()) {
+            throw new IllegalArgumentException("Agent name must be different than null and not empty");
+        }
+        if(entrypoint.getPath() == null || entrypoint.getPath().isEmpty()) {
+            throw new IllegalArgumentException("Path must be different than null and not empty");
+        }
+        entrypoints.get(name).add(entrypoint);
+        LOGGER.info(entrypoints.toString());
+    }
+
+    @RequestMapping(value = "/container/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public Set<String> listContainers() {
+        return entrypoints.keySet();
+    }
+
+    @RequestMapping(value = "/{name}/entrypoint/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
we do not have the entrypoint granularity anymore for entrypoints, will be shown in the next review